packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -968,6 +994,7 @@ function WPLineItem( {
 	isSummary?: boolean;
 	createUserAndSiteBeforeTransaction?: boolean;
 	responseCart: ResponseCart;
+	selectedSiteData: SiteDetails | undefined | null;
```

Comment:
Because this isn't optional, you're going to end up with TypeScript errors in the other places that render line items, unless you provide `selectedSiteData` in each of them. You could avoid that by making this prop optional, but that means that other places (like the masterbar cart) that render this component won't have the right data they need to render the line item.

You might want to consider if there's another way to get the data that you need... you basically need to know what the current site's plan is, more specifically if the current site's plan is a 100-year plan, or more specifically still if the  domain in the cart is included with the current site's plan. That (especially the last one) seems like information that you could pretty easily provide in the shopping cart endpoint. See the code where we add properties like `is_domain_registration`, `is_bundled`, and `is_gift_purchase`. You could add something like `is_included_in_active_plan`.

If you provide this data in the shopping cart response, you won't need to inject the site data here and the calypso logic becomes much simpler. It could also be useful for other code in the future.